x/epochstorage/keeper/stake_storage.go

```diff
@@ -146,85 +146,90 @@ func (k Keeper) SetStakeStorageCurrent(ctx sdk.Context, chainID string, stakeSto
 		}
 	}
-	return 0, false
-}
 
-func (k Keeper) GetStakeEntryByAddressFromStorage(ctx sdk.Context, stakeStorage types.StakeStorage, address sdk.AccAddress) (value types.StakeEntry, found bool, index uint64) {
-	idx, found := k.stakeEntryIndexByAddress(ctx, stakeStorage, address)
-	if !found {
-		return types.StakeEntry{}, false, 0
-	}
-	// found the right thing
-	value = stakeStorage.StakeEntries[idx]
-	found = true
-	index = idx
-	return
+	return types.StakeEntry{}, false
 }
 
-func (k Keeper) GetStakeEntryByAddressCurrent(ctx sdk.Context, chainID string, address sdk.AccAddress) (value types.StakeEntry, found bool, index uint64) {
-	stakeStorage, found := k.GetStakeStorageCurrent(ctx, chainID)
-	if !found {
-		return types.StakeEntry{}, false, 0
+func (k Keeper) GetStakeEntryByAddressCurrent(ctx sdk.Context, chainID string, address string) (types.StakeEntry, bool) {
+	if !utils.IsBech32Address(address) {
+		utils.LavaFormatWarning("cannot get stake entry of invalid address", fmt.Errorf("invalid address"),
+			utils.LogAttr("address", address),
+		)
 	}
-	// the following finds the address of stakeEntry and returns it
-	idx, found := k.stakeEntryIndexByAddress(ctx, stakeStorage, address)
+	stakeStorage, found := k.GetStakeStorageCurrent(ctx, chainID)
 	if !found {
-		return types.StakeEntry{}, false, 0
+		return types.StakeEntry{}, false
 	}
-	// found the right thing
-	value = stakeStorage.StakeEntries[idx]
-	found = true
-	index = idx
-	return
+
+	return k.GetStakeEntryByAddressFromStorage(ctx, stakeStorage, address)
 }
 
-func (k Keeper) RemoveStakeEntryCurrent(ctx sdk.Context, chainID string, idx uint64) error {
+func (k Keeper) RemoveStakeEntryCurrent(ctx sdk.Context, chainID string, address string) error {
+	if !utils.IsBech32Address(address) {
+		utils.LavaFormatWarning("cannot get stake entry of invalid address", fmt.Errorf("invalid address"),
+			utils.LogAttr("address", address),
+		)
+	}
 	stakeStorage, found := k.GetStakeStorageCurrent(ctx, chainID)
 	if !found {
 		return legacyerrors.ErrNotFound
 	}
-	if idx >= uint64(len(stakeStorage.StakeEntries)) {
-		return legacyerrors.ErrNotFound
+
+	for i, entry := range stakeStorage.StakeEntries {
+		if entry.Address == address {
+			stakeStorage.StakeEntries = append(stakeStorage.StakeEntries[:i], stakeStorage.StakeEntries[i+1:]...)
+			k.SetStakeStorageCurrent(ctx, chainID, stakeStorage)
+			return nil
+		}
 	}
-	stakeStorage.StakeEntries = append(stakeStorage.StakeEntries[:idx], stakeStorage.StakeEntries[idx+1:]...)
-	k.SetStakeStorageCurrent(ctx, chainID, stakeStorage)
-	return nil
+
+	return legacyerrors.ErrNotFound
 }
 
-func (k Keeper) AppendStakeEntryCurrent(ctx sdk.Context, chainID string, stakeEntry types.StakeEntry) {
-	// this stake storage entries are sorted by stake amount
-	stakeStorage, found := k.GetStakeStorageCurrent(ctx, chainID)
+func (k Keeper) AppendStakeEntryFromStorage(ctx sdk.Context, stakeStorageIndex string, stakeEntry types.StakeEntry, isUnstake bool) {
```

Comment:
why do we need this?

Reply:
append stake entry will only happen on the current stake storage

Reply:
I can't do that because I use the keeper